test/Microsoft.ML.Tests/Transformers/PcaTests.cs

```diff
@@ -2,55 +2,69 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.IO;
 using Microsoft.ML.Runtime.Data;
 using Microsoft.ML.Runtime.Data.IO;
 using Microsoft.ML.Runtime.RunTests;
 using Microsoft.ML.Transforms;
-using System.IO;
 using Xunit;
 using Xunit.Abstractions;
 
 namespace Microsoft.ML.Tests.Transformers
 {
     public sealed class PcaTests : TestDataPipeBase
     {
+        private readonly ConsoleEnvironment _env;
+        private readonly string _dataSource;
+        private readonly TextSaver _saver;
+
         public PcaTests(ITestOutputHelper helper)
             : base(helper)
         {
+            _env = new ConsoleEnvironment(seed: 1);
+            _dataSource = GetDataPath("generated_regression_dataset.csv");
+            _saver = new TextSaver(_env, new TextSaver.Arguments { Silent = true, OutputHeader = false });
         }
 
         [Fact]
         public void PcaWorkout()
         {
-            var env = new ConsoleEnvironment(seed: 1, conc: 1);
-            string dataSource = GetDataPath("generated_regression_dataset.csv");
-            var data = TextLoader.CreateReader(env,
-                c => (label: c.LoadFloat(11), features: c.LoadFloat(0, 10)),
+            var data = TextLoader.CreateReader(_env,
```

Comment:
>data [](start = 16, length = 4)

nit: you can create third data with same schema but loading floats from 1 to 6, and pass it to validForFitNotValidForTransformInput

Reply:
Is this what you mean?
c.LoadFloat(11), weight: c.LoadFloat(0), features: c.LoadText(1, 6)
that sounds like validFitInput to me.

---
In reply to: [227574746](https://github.com/dotnet/machinelearning/pull/1333#discussion_r227574746) [](ancestors = 227574746)

Reply:
if you replace features:c.LoadText(1,6) to features:c.LoadFloat(1,6) then yes (I doubt you can fit on top of text vector)
And that's exactly my point, it's a valid fit but if you already train you data on bigger array, it should fail on smaller one, this is why it called validForFitNotValidForTransformInput :)

---
In reply to: [227626841](https://github.com/dotnet/machinelearning/pull/1333#discussion_r227626841) [](ancestors = 227626841,227574746)